### PR TITLE
Athena Neptune Connector support for Custom query

### PR DESCRIPTION
**Pull request by abhishekpradeepmishra**
*Issue #, if available:*
https://github.com/awslabs/aws-athena-query-federation/issues/785

https://github.com/awslabs/aws-athena-query-federation/issues/775

*Description of changes:*

Added support for custom query feature

You can now create a custom schema and specify a gremlin query to fetch the data for the schema from the graph databases. This enhancement allows users to specify a gremlin query instead of doing table join operations to retrieve data. You can also put a limit on number of records you want to retrieve.

Refer doc: https://github.com/abhishekpradeepmishra/aws-athena-query-federation/blob/custom-query/athena-neptune/docs/aws-glue-sample-scripts/README.md

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by avirtuos** [comment]
Sorry for the long turn around time. I went through the change and it seems reasonable to me. The only concern I have is around how we are using Glue Catalog to define what is essentially a "view" over the Neptune data database. I've asked @burhan94 to take a look at this PR as well since I vaguely recall Glue had a "best practice" if not something even more formal for how to define a view. Would be nice to use that approach (which might just be different property names from what you are already doing).

**Review by avirtuos — APPROVED** [review]
Looks Good To Me, though I did post 1 question about how best to define what is effectively a view in data catalog. @burhan94 will also take a look before we merge this.

**Comment by avirtuos** [comment]
I'm approving the workflow so we get the build and tests to run but I'm going to hold off on merging until we get @burhan94's feedback in.

**Comment by burhan94** [comment]
Hi Abishek, i'll ping you separately, but we're planning on launching a feature later this year where customers will be able to do something similar as to whats in this PR. I'll review the PR later today, but wanted to call out that there will be an opportunity to simplify this feature by EOY.

**Comment by abhishekpradeepmishra** [comment]
Hi, kindly let now if some changes are needed. And agree when the feature is enabled we should simply this

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java
Comment (abhishekpradeepmishra) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java:146]: will make changes

File: athena-neptune/pom.xml
Comment (abhishekpradeepmishra) [reply athena-neptune/pom.xml:12]: it was breaking for older neptune version, let me check again once

File: athena-neptune/docs/aws-glue-sample-scripts/automation/script.js
Comment (abhishekpradeepmishra) [reply athena-neptune/docs/aws-glue-sample-scripts/automation/script.js:32]: will fix

File: athena-neptune/pom.xml
Comment (abhishekpradeepmishra) [reply athena-neptune/pom.xml]: Done

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java
Comment (abhishekpradeepmishra) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java]: Done

File: athena-neptune/docs/aws-glue-sample-scripts/automation/script.js
Comment (abhishekpradeepmishra) [reply athena-neptune/docs/aws-glue-sample-scripts/automation/script.js]: Done

**Comment by abhishekpradeepmishra** [comment]
Changed componenttype to view and made other changes. Kindly review

**Comment by abhishekpradeepmishra** [comment]
Kindly approve if no further needed

**Review by burhan94 — APPROVED** [review]

**Pull request merged**
